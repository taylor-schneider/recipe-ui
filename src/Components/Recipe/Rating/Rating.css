div.Rating {
    width: 100%;
    height: max-content;
    background-color: antiquewhite;
}
div.Rating > span {
    height: 4vh;
}
div.Rating > span > img.star-img {
    display: inline-block;
    height: 4vh;
}
div.Rating > span:nth-child(1){
    padding-right: 1vw;    
}
div.Rating > span:nth-child(2){
    display: inline-block;
    line-height: 4vh;
}