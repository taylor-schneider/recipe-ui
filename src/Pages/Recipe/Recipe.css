div.Recipe {
    height: 100%;
}
