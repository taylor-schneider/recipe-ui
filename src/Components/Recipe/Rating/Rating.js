import './Rating.css';
import star_img from './star.png'

function Rating(recipe) {

  console.log("star_img")
  console.log(star_img)

  var star_imgs = []
  var reviews = 0
  var reviews_text = ""

  if(recipe != null){

    // Create the rating image
    for(let i = 0; i < recipe.rating; i++){
      star_imgs.push(<img className='star-img' src={star_img}></img>)
    }

    // Determine how people reviewed the recipe
    if(recipe.reviews != null){
      reviews = recipe.reviews.length
      if (reviews == 1) {
        reviews_text = "(" + reviews + " Reviews)"
      }
      else{
        reviews_text = "(" + reviews + " Reviews)"
      }
    }
  }


  return (
    <div className="Rating">
      <span>{star_imgs}</span>
      <span>{reviews_text}</span>  
    </div>
  );
}

export default Rating;
