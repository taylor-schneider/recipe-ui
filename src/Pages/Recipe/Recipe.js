import './Recipe.css';
import React, { Component, useEffect } from "react";
import {useLocation, useNavigate, useParams } from "react-router-dom";
import DefaultLayout from '../../Layouts/Default/Default';


// Import generic contiainers
import Section from '../../Sections/Section/Section';
import Panel from '../../Panels/Panel/Panel';

// Import relevant components
import Title from '../../Components/Recipe/Title/Title'
import Rating from '../../Components/Recipe/Rating/Rating'
import Authors from '../../Components/Recipe/Authors/Authors'
import Description from '../../Components/Recipe/Description/Description'
import ImageGallery from '../../Components/Recipe/ImageGallery/ImageGallery';
import PanelScroller from '../../Panels/PanelScroller/PanelScroller';
import SectionScroller from '../../Sections/SectionScroller/SectionScroller';
import Version from '../../Components/Recipe/Version/Version';


class Recipe extends Component {

  state = {
    recipe: {}
  };

  componentDidMount() {

    // Do some logic to extract the uri params
    const search_string = this.props.router.location.search
    const queryParams = new URLSearchParams(search_string)
    const guid = queryParams.get("guid")
    // Determine the url to the api based on the params
    const version = queryParams.get("version")
    var recipe_url = "http://127.0.0.1:8008/recipe?guid=" + guid;
    if (version != null){
      recipe_url += "&version=" + version
    }
    // Query the api for the specified recipe and set the state for the component
    fetch(recipe_url)
      .then(res => res.json())
      .then((recipe) => {
        this.setState({ recipe: recipe })
      });
  }

  render() {

    return (
      <DefaultLayout>
        <div className="Recipe">
        </div>
      </DefaultLayout>
    );
  }
}

//          
//

function withRouter(Component) {
  function ComponentWithRouterProp(props) {
    let location = useLocation();
    let navigate = useNavigate();
    let params = useParams();
    return (
      <Component
        {...props}
        router={{ location, navigate, params }}
      />
    );
  }

  return ComponentWithRouterProp;
}

export default withRouter(Recipe)