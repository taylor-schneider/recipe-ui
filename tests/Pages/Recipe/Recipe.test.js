import Recipe from '../../../src/Pages/Recipe/Recipe'
import Rating from '../../../src/Components/Recipe/Rating/Rating'
import renderer from "react-test-renderer";

var recipe_dict = {
    "guid": "foobar",
    "name": "Baz's Famous Maccaroni And Cheese",
    "version": "v1.1.1",
    "description": "This is a tasty dish that kept baz alive in college.",
    "authors": [
        "@Shebaz",
        "@some-other-guy"
    ],
    "nutrition_facts": {
        "serving_info": {
            "servings":"6",
            "serving_size": "6 bowls"
        },
    },
    "prep_time": {
        "quantity": 45,
        "unit": "min"
    },
    "rating": "5",
    "based_on": null,
    "reviews": [
        {
            "author": "@Jim",
            "comment": "It was dank!",
            "rating": 4.5
        }
    ],
    "tags": [],
    "classifications": {},
    "ingredients": [
        {
            "name": "shreded cheese",
            "quantity": "1",
            "unit": "cup"
        },
        {
            "name": "noddles",
            "quantity": "1",
            "unit": "package"
        }
    ],
    "tools": [
        "stove top",
        "pot",
        "spoon"
    ],
    "instructions": [
        {
            "description": "Boil the noodles.",
            "tools": [0, 1],
            "ingredients":[0]
        },
        {
            "description": "Put the cheese on the noodles.",
            "tools": [1],
            "ingredients":[0, 1]
        },
        {
            "description": "Stir the noodles an the cheese.",
            "tools": [1, 2],
            "ingredients":[0, 1]
        }
    ]
}

describe("When loading a rating", () => {
    test('It renders properly', () => {

        // Create the component
        var rating = Rating(recipe_dict)

        // Create the DOM tree of the rendered component in JSON format.
        const domTree = renderer.create(rating).toJSON();

        // Create a snapshot if it does not exist, save it, and check if the snapshot is 
        // consistent with previous stored snapshots. If there is an existing snapshot, 
        // Jest compares the two snapshots. If they match, the test passes. Snapshots that 
        // do not match cause the test to fail.
        expect(domTree).toMatchSnapshot();
    });
   })