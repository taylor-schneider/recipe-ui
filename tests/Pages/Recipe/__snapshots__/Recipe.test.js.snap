// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When loading a rating It renders properly 1`] = `
<div
  className="Rating"
>
  <span>
    <img
      className="star-img"
      src="/public/9261fb9b46f96056791546b787d10669.png"
    />
    <img
      className="star-img"
      src="/public/9261fb9b46f96056791546b787d10669.png"
    />
    <img
      className="star-img"
      src="/public/9261fb9b46f96056791546b787d10669.png"
    />
    <img
      className="star-img"
      src="/public/9261fb9b46f96056791546b787d10669.png"
    />
    <img
      className="star-img"
      src="/public/9261fb9b46f96056791546b787d10669.png"
    />
  </span>
  <span>
    (1 Reviews)
  </span>
</div>
`;
